src/audio/wav64_opus.c

```diff
@@ -97,17 +91,19 @@ static void waveform_opus_read(void *ctx, samplebuffer_t *sbuf, int wpos, int wl
             memset(out, 0, st->xhead.frame_size * wav->wave.channels * sizeof(int16_t));
         } else {
             // Read frame size
-            int nb = io_read16(st->current_rom_addr);
+            int nb;
+            read(st->currentfile, &nb, 2);
             assertf(nb <= st->xhead.max_cmp_frame_size, "opus frame size too large: %d (%ld)", nb, st->xhead.max_cmp_frame_size);
 
             // Read frame
             data_cache_hit_writeback_invalidate(buf, nb);
-            dma_read(buf, st->current_rom_addr + 2, nb);
+            read(st->currentfile, buf, nb);
 
             // Update ROM pointer after current frame
-            st->current_rom_addr += nb + 2;
-            if (st->current_rom_addr & 1)
-                st->current_rom_addr += 1;
+            int currentStreamOffset = lseek(st->currentfile, 0, SEEK_CUR);
+            if (currentStreamOffset & 1) {
+                lseek(st->currentfile, 1, SEEK_CUR);
+            }
```

Comment:
The file position is always 2-byte aligned here at the point that the frame begins:
https://github.com/DragonMinded/libdragon/blob/7ce2ab0a487de6c9234c619a0d2ae6a6943afdb3/tools/audioconv64/conv_wav64.c#L401

In other words, `nb` always starts at an even offset. So basically the idea here is that you must skip 1 byte only if `nb` is odd. Or even better, to save the `lseek` call, you could simply read 1 more byte in the read() when `nb` is odd (obviously make sure to increase `but` by one byte in that case).

Reply:
I can get rid of the first 1st lseek and check (nb & 1) instead. Nice optimization!
For the second one Using lseek to advince the fileptr sound like a better solution here right? Otherwise read would compile to a load instruction that may cause a cache miss and have to go over the PI bus for nothing. Also have to allocate some stack space for a read.

Reply:
It's just one additional byte. PI bus is 16 bit anyway, so if you request an odd number of bytes, one more byte is transferred anyway. lseek is fast but can't be faster than just reading a byte. If you're not convinced, you could measure it with `get_ticks()`, I might be wrong.